Add an Access Request configuration guide

ptgott opened the pull request:
Fixes #22496
Fixes #13927
Fixes #6557
Fixes #31095
Fixes #29980
Fixes #17630

The current approach to documenting Access Requests is to include how-to guides for various scenarios. The downside of this approach is that we don't have a great place to put general conceptual discussions of Access Request configuration fields. This leads to confusion among users regarding the way Teleport handles certain Access Request configuration options.

This change adds a conceptual guide that explains all of the fields in a Teleport role that are relevant to configuring Access Requests.

This change also moves conceptual discussions from other guides into the new guide. To limit the scope of this change, it is not intended to overhaul the existing guides. If a discussion of a particular configuration field was buried in another guide, this change moves it into the new guide:
- `preview_as_roles` discussion in the Role Requests guide
- TTL information in the Role Requests guide
- The reference role in the Role Requests page

ptgott committed:
Add an Access Request configuration guide

Fixes #22496
Fixes #13927
Fixes #6557
Fixes #31095
Fixes #29980
Fixes #17630

The current approach to documenting Access Requests is to include how-to
guides for various scenarios. The downside of this approach is that we
don't have a great place to put general conceptual discussions of Access
Request configuration fields. This leads to confusion among users
regarding the way Teleport handles certain Access Request configuration
options.

This change adds a conceptual guide that explains all of the fields in a
Teleport role that are relevant to configuring Access Requests.

This change also moves conceptual discussions from other guides into the
new guide. To limit the scope of this change, it is not intended to
overhaul the existing guides. If a discussion of a particular
configuration field was buried in another guide, this change moves it
into the new guide:
- `preview_as_roles` discussion in the Role Requests guide
- TTL information in the Role Requests guide
- The reference role in the Role Requests page

zmb3 commented on docs/pages/access-controls/access-requests/access-request-configuration.mdx:
```diff
@@ -0,0 +1,622 @@
+
+### The `request.max_duration` field
+
+The `max_duration` option indicates the maximum length of time that a user is
+allowed to have elevated privileges for particular roles. You can specify it
```
I would consider mentioning `max_duration` towards the end, since it's basically an exception to the default rules.

I might also clarify that a `max_duration` of `4d` doesn't mean you can have continuous access for 4 days - you're still limited by the session TTL, but you can re-assume the access to restart the session TTL clock for up to 4 days.

zmb3 commented on docs/pages/access-controls/access-requests/access-request-configuration.mdx:
```diff
@@ -0,0 +1,622 @@
+Teleport uses the following logic to determine how long an Access Request is
+valid while it awaits approval:
+
+1. Begin with with the base expiration of the Access Request, which a user can
```
I believe @atburke _just_ added this to the web UI too (a day or two ago - may not even be released yet)

zmb3 commented on docs/pages/access-controls/access-requests/access-request-configuration.mdx:
````diff
@@ -0,0 +1,622 @@
+
+```yaml
+kind: role
+version: v6
````
Would it make sense to use the same version for all the roles on this page? Otherwise people might wonder if the discrepancy is intentional..

zmb3 commented on docs/pages/access-controls/access-requests/access-request-configuration.mdx:
```diff
@@ -0,0 +1,622 @@
+|`contains([]string, val)`|Returns `true` if the list of strings in the first argument contains the element in the second argument and `false` otherwise.|
+|`expr1 && expr2`|Evaluates to `true` if both `expr1` and `expr2` evaluate to `true`.|
+|`expr1 \|\| expr2`|Evaluates to `true` if `expr1`, `expr2`, or both evaluate to `true`.|
+|`!expr`|Evaluates to the opposite of `expr`.|
```
```suggestion
|`!expr`|Negates `expr`.|
```

zmb3 reviewed (approved):
This is a massive improvement! Thanks for taking it on and please treat my comments as suggestions that do not block merge.

There were 2-3 things in here that I am just learning about for the first time!

ptgott committed:
Respond to zmb3 feedback

- Use clearer wording when describing key concepts.
- Include more detail about the `max_duration` field.
- Use consistent version numbers in example roles.

ptgott committed:
Fix linter issues
